frontend/packages/console-plugin-sdk/src/store.ts

```diff
@@ -155,8 +233,21 @@ type DynamicPluginManifest = Readonly<ConsolePluginManifestJSON>;
 
 type DynamicPluginMetadata = Omit<DynamicPluginManifest, 'extensions'>;
 
-type DynamicPlugin = {
+type LoadedDynamicPlugin = {
   manifest: DynamicPluginManifest;
   processedExtensions: Readonly<LoadedExtension[]>;
   enabled: boolean;
 };
+
+type LoadedDynamicPluginInfo = DynamicPluginMetadata & {
+  status: 'Loaded';
+  pluginID: string;
+  enabled: boolean;
+};
+
+type NotLoadedDynamicPluginInfo = {
+  status: 'Pending' | 'Failed';
+  pluginName: string;
+};
+
+export type DynamicPluginInfo = LoadedDynamicPluginInfo | NotLoadedDynamicPluginInfo;
```

Comment:
Seeing the following TypeScript errors:
```
ERROR in /Users/rhamilto/Git/console/frontend/packages/operator-lifecycle-manager/src/components/console-operator-config.tsx(51,54):
TS2339: Property 'name' does not exist on type 'DynamicPluginInfo'.
  Property 'name' does not exist on type 'NotLoadedDynamicPluginInfo'.
ERROR in /Users/rhamilto/Git/console/frontend/packages/operator-lifecycle-manager/src/components/console-operator-config.tsx(54,24):
TS2339: Property 'version' does not exist on type 'DynamicPluginInfo'.
  Property 'version' does not exist on type 'NotLoadedDynamicPluginInfo'.
ERROR in /Users/rhamilto/Git/console/frontend/packages/operator-lifecycle-manager/src/components/console-operator-config.tsx(55,28):
TS2339: Property 'description' does not exist on type 'DynamicPluginInfo'.
  Property 'description' does not exist on type 'NotLoadedDynamicPluginInfo'.
```
with the following code:
```
const ConsolePluginsList: React.FC<ConsolePluginsListType> = ({ obj }) => {
  const { t } = useTranslation();
  const [consolePlugins, consolePluginsLoaded] = useK8sWatchResource<ConsolePluginKind[]>({
    isList: true,
    kind: referenceForModel(ConsolePluginModel),
  });
  const dynamicPluginInfo: DynamicPluginInfo[] = useDynamicPluginInfo();
  const [rows, setRows] = React.useState([]);
  const [sortBy, setSortBy] = React.useState<ISortBy>({});
  React.useEffect(() => {
    const data = consolePlugins.map((plugin) => {
      const info = dynamicPluginInfo?.find((i) => i?.name === plugin?.metadata?.name);
      return {
        name: plugin?.metadata?.name,
        version: info?.version,
        description: info?.description,
        status: info?.status,
      };
    });
    setRows(
      data?.map((item) => {
        const placeholder = '-';
        return [
          item.name,
          item.version || placeholder,
          item.description || placeholder,
          item?.status || placeholder,
        ];
      }),
    );
  }, [consolePlugins, dynamicPluginInfo]);
...
```
Something is incorrect with the `type` or I am using it incorrectly?

Reply:
@rhamilto This is because the overlap of type union `LoadedDynamicPluginInfo | NotLoadedDynamicPluginInfo` is just the `status` field.

I've updated the PR and added `isLoadedDynamicPluginInfo` type guard which narrows `DynamicPluginInfo` down to `LoadedDynamicPluginInfo` for easier consumption.

```ts
const dynamicPluginInfo = useDynamicPluginInfo();

// inside consolePlugins.map() transformer
const loadedPluginInfo = dynamicPluginInfo
  .filter(isLoadedDynamicPluginInfo)
  .find((i) => i.name === plugin.metadata.name);

return {
  name: plugin.metadata.name,
  version: loadedPluginInfo?.version || '<not available>',
  description: loadedPluginInfo?.description || '<not available>',
};
```